test/unit/helpers/events.ts

```diff
@@ -59,13 +59,14 @@ const ack: AckFn<void> = (_r?) => Promise.resolve();
 export function wrapMiddleware<Args extends AnyMiddlewareArgs>(
   args: Args,
   ctx?: Context,
-): Args & AllMiddlewareArgs & { next: SinonSpy } {
+): Args & AllMiddlewareArgs & { next: SinonSpy; ack: SinonSpy } {
```

Comment:
I don't think this is correct. `Args` here extends `AnyMiddlewareArgs` which is a union of different kinds of middleware args. _Some_ of these middleware args contain `ack` and others do not, so in theory `Args` should be carrying the correct type, including the conditional presence of `ack`.

You may have to narrow the return type from this method appropriately to get the right kind of `Args` depending on the situation, which should in theory sometimes carry `ack`.

I assume the issue was asserting on `ack` in the test file you changed in this PR?